include/rcutils/logging.h

```diff
@@ -116,45 +181,163 @@ rcutils_logging_output_handler_t rcutils_logging_get_output_handler();
+ * ------------------ | -------------
+ * Allocates Memory   | No, provided logging system is already initialized
+ * Thread-Safe        | No
+ * Uses Atomics       | No
+ * Lock-Free          | Yes
+ *
+ * \param name The name of the logger, must be null terminated c string
+ * \return The severity threshold of the logger if it has been set, or
+ * \return `RCUTILS_LOG_SEVERITY_UNSET` if unset, or
+ * \return `g_rcutils_logging_default_severity_threshold` for an empty name, or
+ * \return -1 on invalid arguments, or
+ * \return -1 if an error occurred
+ */
+RCUTILS_PUBLIC
+int rcutils_logging_get_logger_severity_threshold(const char * name);
+
+/// Get the severity threshold for a logger and its name length.
+/**
+ * Identical to rcutils_logging_get_logger_severity_threshold() but without
+ * relying on the logger name to be a null terminated c string.
+ *
+ * <hr>
+ * Attribute          | Adherence
+ * ------------------ | -------------
+ * Allocates Memory   | No, provided logging system is already initialized
+ * Thread-Safe        | No
+ * Uses Atomics       | No
+ * Lock-Free          | Yes
+ *
+ * \param name The name of the logger
+ * \param name_length Logger name length
+ * \return The severity threshold of the logger if it has been set, or
+ * \return `RCUTILS_LOG_SEVERITY_UNSET` if unset, or
+ * \return `g_rcutils_logging_default_severity_threshold` for `name_length` of `0`, or
+ * \return -1 on invalid arguments, or
+ * \return -1 if an error occurred
+ */
+RCUTILS_PUBLIC
+int rcutils_logging_get_logger_severity_thresholdn(const char * name, size_t name_length);
+
+/// Set the severity threshold for a logger.
+/**
+ * If an empty string is specified as the name, the
+ * `g_rcutils_logging_default_severity_threshold` will be set.
+ *
+ * <hr>
+ * Attribute          | Adherence
+ * ------------------ | -------------
+ * Allocates Memory   | Yes
+ * Thread-Safe        | No
+ * Uses Atomics       | No
+ * Lock-Free          | Yes
+ *
+ * \param name The name of the logger, must be null terminated c string.
+ * \param severity The severity threshold to be used.
+ * \return `RCUTILS_RET_OK` if successful, or
+ * \return `RCUTILS_RET_INVALID_ARGUMENT` on invalid arguments, or
+ * \return `RCUTILS_RET_LOGGING_SEVERITY_MAP_INVALID` if severity map invalid, or
+ * \return `RCUTILS_RET_ERROR` if an unspecified error occured
+ */
+RCUTILS_PUBLIC
+rcutils_ret_t rcutils_logging_set_logger_severity_threshold(const char * name, int severity);
+
+/// Determine if a logger is enabled for a severity.
+/**
+ * <hr>
+ * Attribute          | Adherence
+ * ------------------ | -------------
+ * Allocates Memory   | No, provided logging system is already initialized
+ * Thread-Safe        | No
+ * Uses Atomics       | No
+ * Lock-Free          | Yes
+ *
+ * \param name The name of the logger, must be null terminated c string or NULL.
+ * \param severity The severity level.
+ *
+ * \return True if the logger is enabled for the severity; false otherwise.
```

Comment:
nitpick: `True` is uppercase and `false` is lower, lower is correct in C++

Reply:
thanks, I was capitalising the sentence. fixed in 302466762b2c7b3419536f2196100a502a1e44c9